Hospital.Infrastructure.EF/Repositories/AppointmentEfRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Hospital.Domain.Entities;
+using Hospital.Domain.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace Hospital.Infrastructure.EF.Repositories;
+
+/// <summary>
+/// Entity Framework Core repository implementation for managing <see cref="Appointment"/> entities.
+/// Provides CRUD operations using the <see cref="HospitalDbContext"/>.
+/// </summary>
+public class AppointmentEfRepository(HospitalDbContext db) : IAppointmentRepository
+{
+    /// <summary>
+    /// Retrieves all appointments from the database, including related <see cref="Patient"/> and <see cref="Doctor"/> entities.
+    /// </summary>
+    /// <returns>
+    /// A collection of <see cref="Appointment"/> objects ordered by their identifier.
+    /// </returns>
+    public IEnumerable<Appointment> GetAll() =>
+        db.Appointments
+          .Include(a => a.Patient)
+          .Include(a => a.Doctor)
+          .AsNoTracking()
+          .OrderBy(a => a.Id)
+          .ToList();
```

Comment:
collection expression

```suggestion
    public IEnumerable<Appointment> GetAll() =>
        [.. db.Appointments
          .Include(a => a.Patient)
          .Include(a => a.Doctor)
          .AsNoTracking()
          .OrderBy(a => a.Id)];
```